File: app/src/main/java/com/ballastlane/android/texasam/app/ui/init/splash/ActivitySplash.java
package com.ballastlane.android.texasam.app.ui.init.splash;

import android.content.Intent;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.v7.app.AppCompatDelegate;

import com.ballastlane.android.baselibrary.common.base.BaseActivity;
import com.ballastlane.android.texasam.R;
import com.ballastlane.android.texasam.app.ui.init.ActivityInit;
import com.ballastlane.android.texasam.app.ui.main.ActivityMain;
import com.ballastlane.android.texasam.databinding.ActivitySplashBinding;
import com.ballastlane.android.texasam.utils.PreferencesUtils;

import java.util.concurrent.TimeUnit;

import io.reactivex.Observable;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.functions.Consumer;
import io.reactivex.schedulers.Schedulers;

/**
 * Created by Julian Cardona on 6/23/16.
 */
public class ActivitySplash extends BaseActivity {

    static {
        AppCompatDelegate.setCompatVectorFromResourcesEnabled(true);
    }

    private static final int DEFAULT_DELAY_TIME = 3;

    private ActivitySplashBinding binding;


    @Override
    protected void initVars() {
        PreferencesUtils.getInstance().setUserLogged(false);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        this.binding = DataBindingUtil.setContentView(this, R.layout.activity_splash);
    }

    @Override
    protected void initViews() {
        startWaitingTime();
    }


    @Override
    protected void initListeners() {

    }

    private void startWaitingTime() {
        addDisposable(
                Observable
                        .just(true)
                        .delay(DEFAULT_DELAY_TIME, TimeUnit.SECONDS)
                        .subscribeOn(Schedulers.io())
                        .observeOn(AndroidSchedulers.mainThread())
                        .subscribe(
                                new Consumer<Boolean>() {
                                    @Override
                                    public void accept(Boolean aBoolean) throws Exception {
                                        startActivity(new Intent(ActivitySplash.this,
                                                PreferencesUtils.getInstance().isUserLogged() ?
                                                        ActivityMain.class :
                                                        ActivityInit.class));
                                    }
                                }
                                ,
                                new Consumer<Throwable>() {
                                    @Override
                                    public void accept(Throwable throwable) throws Exception {

                                    }
                                }
                        )
        );
    }

}

File: sample/src/main/java/com/ballastlane/ballastlanebaseapp/app/ui/splash/ActivitySplash.kt
package com.ballastlane.ballastlanebaseapp.app.ui.splash

import android.os.Bundle
import com.ballastlane.android.baselibrary.common.base.BaseActivity

/**
 * Created by Mariangela Salcedo (mariangelasalcedo@ballastlane.com) on 4/2/18.
 * Copyright (c) 2018 Ballast Lane Applications LLC. All rights reserved.
 */

class ActivitySplash : BaseActivity() {
    override fun initViews() {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun initListeners() {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }
    //  private var binding: ActivitySplashBinding? = null

    override fun initVars() {}

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
//        this.binding = DataBindingUtil.setContentView<ActivitySplashBinding>(this, R.layout.activity_splash)
    }

//    override fun initViews() {
//        startLogoAnimation()
//        AddAnimationBlink()
//        startWaitingTime()
//    }
//
//    override fun initListeners() {}
//
//    private fun startLogoAnimation() {
//        (binding!!.logo.getDrawable() as Animatable).start()
//    }
//
//    private fun AddAnimationBlink() {
//        addDisposable(
//                Observable
//                        .interval(DEFAULT_BLINK_DELAY_TIME.toLong(), TimeUnit.SECONDS)
//                        .timeInterval()
//                        .subscribeOn(Schedulers.io())
//                        .observeOn(AndroidSchedulers.mainThread())
//                        .subscribe(
//                                {
//                                    if (!(binding!!.logo.getDrawable() as Animatable).isRunning) {
//                                        (binding!!.logo.getDrawable() as Animatable).start()
//                                    }
//                                }
//                        ) { }
//        )
//    }
//
//    private fun startWaitingTime() {
//        addDisposable(
//                Observable
//                        .just(true)
//                        .delay(DEFAULT_DELAY_TIME.toLong(), TimeUnit.SECONDS)
//                        .subscribeOn(Schedulers.io())
//                        .observeOn(AndroidSchedulers.mainThread())
//                        .subscribe(
//                                { startActivity(Intent(this@ActivitySplash, ActivityMain::class.java)) }
//                        ) { }
//        )
//    }
//
//    companion object {
//
//        init {
//            AppCompatDelegate.setCompatVectorFromResourcesEnabled(true)
//        }
//
//        private val DEFAULT_BLINK_DELAY_TIME = 4
//        private val DEFAULT_DELAY_TIME = 5
//    }

}